File: app/src/main/java/io/dwak/holohackernews/app/network/HackerNewsService.java
package io.dwak.holohackernews.app.network;

import java.util.List;

import io.dwak.holohackernews.app.network.models.NodeHNAPIStory;
import io.dwak.holohackernews.app.network.models.NodeHNAPIStoryDetail;
import retrofit.Callback;
import retrofit.http.GET;
import retrofit.http.Path;

/**
 * Retrofit service to interface with hacker news api
 * Created by vishnu on 4/21/14.
 */
public interface HackerNewsService {
    @GET("/news")
    void getTopStories(Callback<List<NodeHNAPIStory>> callback);

    @GET("/news2")
    void getTopStoriesPageTwo(Callback<List<NodeHNAPIStory>> callback);

    @GET("/newest")
    void getNewestStories(Callback<List<NodeHNAPIStory>> callback);

    @GET("/best")
    void getBestStories(Callback<List<NodeHNAPIStory>> callback);

    @GET("/item/{itemId}")
    void getItemDetails(@Path("itemId") long itemId, Callback<NodeHNAPIStoryDetail> callback);
}

File: app/src/main/kotlin/io/dwak/holohackernews/app/network/HackerNewsService.kt
package io.dwak.holohackernews.app.network

import io.dwak.holohackernews.app.model.json.StoryDetailJson
import io.dwak.holohackernews.app.model.json.StoryJson
import retrofit2.http.GET
import retrofit2.http.Path
import rx.Observable

interface HackerNewsService {
    @GET("/news") fun getTopStories() : Observable<MutableList<StoryJson>>
    @GET("/news2") fun getTopStoriesPageTwo() : Observable<MutableList<StoryJson>>
    @GET("/newest") fun getNewestStories() : Observable<MutableList<StoryJson>>
    @GET("/best") fun getBestStories() : Observable<MutableList<StoryJson>>
    @GET("/show") fun getShowHnStories() : Observable<MutableList<StoryJson>>
    @GET("/shownew") fun getNewShowHnStories() : Observable<MutableList<StoryJson>>
    @GET("/ask") fun getAskHnStories() : Observable<MutableList<StoryJson>>
    @GET("/item/{itemId}") fun getItemDetails(@Path(value = "itemId") itemId : Long) : Observable<StoryDetailJson>
}